BuiltInSpicyAnalyzer.cmake

```
@@ -38,8 +38,9 @@ function (spicy_add_analyzer)
             COMMENT "Compiling ${SPICY_ANALYZER_NAME} analyzer"
             COMMAND
                 ${CMAKE_COMMAND} -E env
-                "ZEEK_SPICY_LIBRARY_PATH=${PROJECT_SOURCE_DIR}/scripts/spicy" $<TARGET_FILE:spicyz>
-                -L ${spicy_SOURCE_DIR}/hilti/lib -L ${spicy_SOURCE_DIR}/spicy/lib -x
+                "ZEEK_SPICY_LIBRARY_PATH=${PROJECT_SOURCE_DIR}/scripts/spicy"
+                ASAN_OPTIONS=$ENV{ASAN_OPTIONS}:detect_leaks=0 $<TARGET_FILE:spicyz> -L
```

```suggestion
                ASAN_OPTIONS=$$ASAN_OPTIONS:detect_leaks=0 $<TARGET_FILE:spicyz> -L
```

Think this is a tiny bit better as otherwise it'll use the ASAN_OPTIONS set at configure time and put the hard-coded into the generated build files which would be surprising (IMO anyway).

---

Is this syntax actually documented to work when using a generator other than `Unix Makefiles`? A search for `$$` in the upstream docs turned up nothing, but maybe I didn't look in the right place?

---

> Is this syntax actually documented to work when using a generator other than `Unix Makefiles`? A search for `$$` in the upstream docs turned up nothing, but maybe I didn't look in the right place.

:sweat_smile:  - it does seem to work for ninja, too, but realize that if it's generator dependent that's not better. Ignore it.
```
/usr/bin/cmake -E env VAL=$VAL bash -c 'set -x; echo Hey $RANDOM val=$VAL
```